actor/src/main/scala-2.12/org/apache/pekko/util/ByteString.scala

```diff
@@ -827,6 +827,15 @@ sealed abstract class ByteString extends IndexedSeq[Byte] with IndexedSeqOptimiz
    */
   def toArrayUnsafe(): Array[Byte] = toArray
 
+  /**
+   * Return the bytes in this ByteString as an InputStream.
+   *
+   * @return the bytes in this ByteString accessible as an InputStream
+   * @see [[asByteBuffer]]
+   * @since 1.1.0
+   */
+  final def asInputStream: InputStream = new ByteArrayInputStream(toArrayUnsafe())
```

Comment:
The `toArrayUnsafe` documentation states:

> If the ByteString is backed by a single array it is returned without any copy. If it is backed by a rope
of multiple ByteString instances a new array will be allocated and the contents will be copied
into it before returning it

So we could avoid that allocation either by:
* defining `def asInputStream: InputStream` as an abstract method, and add different implementations that avoid allocations 
  * using `new ByteArrayInputStream(bytes)` for `ByteString1C`, 
  * using `new ByteArrayInputStream(bytes, start, length)` for `ByteString1`, 
  * using `new SequenceInputStream(bytestrings.map(_.asInputStream))` for `ByteStrings`
* Implement in the super class using `SequenceInputStream(asByteBuffers.map(bb => new ByteBufferBackedInputStream(bb))`

WDYT?

Reply:
Was about to write the same, this should be the implementation

Reply:
good point - I have committed a change based on @jtjeferreira's suggestions